BEIMA.Client/src/service/GetDeviceList.js

```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+const API_URL = process.env.REACT_APP_API_URL;
+
+/**
+ * GET API call to get a list of devices in the database and filters which fields are returned
+ *
+ * @return JSON with list of all devices or error message on failure
+ */
+const GetDeviceList = async() => {
+  const deviceListCall = await axios.get("http://localhost:7071/api/device_list").catch(function (error) {
+      if (error.response) {
+        return error.response;
+    }
+  });
+
+  let filteredFields = deviceListCall.data.map((item) => {
+    return {
+      id: item._id,
+      deviceType: item.deviceTypeId,
+      buildingId: item.location.buildingId,
```

Comment:
I know there is other data such as location.latitude, location.longitude, and location.notes, as well as lastModified.user. Should we consider including some of these? Also, I think I should note that the post endpoint expects location and lastModified data to be nested objects. But I am guessing that this is more intended for use as some top level device data for populating the list row information.

Reply:
Yes, the function is intended to be called to populate the device list page, and I figured these fields were the most important. We can add more if we feel other fields are also important